h/search/util.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+from h._compat import urlparse
+
+
+def wildcard_uri_is_valid(wildcard_uri):
+    """
+    Return True if uri contains wildcards in appropriate places, return False otherwise.
+
+    *'s and _'s are not permitted in the scheme or netloc aka:
+        scheme://netloc/path;parameters?query#fragment.
+
+    If a wildcard is near the begining of a url, elasticsearch will find a large portion of the
+    annotations because it is based on luncene which searches from left to right. In order to
```

Comment:
I realize this comment was just moved from where it was before, but I I'm not sure this explanation is based on the right understanding. Also there is a typo "luncene". The problem is that Lucene is based on inverted indexes and wildcard queries work by expanding the input query to existing terms in the index. The more terms that match, the more expensive the query is to process.  Allowing wildcards near the start of URI terms makes it easy to construct (relatively) very expensive queries.